### ISSUE TITLE
Add Oracle com.oracle.database.security:oraclepki library

### ISSUE THREAD

**Issue by dsyayu** [dbeaver/dbeaver#22312]
### Description

Dear DBeaver Support Team,

I have a requirement to establish a connection to AWS RDS Oracle using port 2484 (SSL) and cwallet.sso. However, I encountered the following error:

Error:
ORA-17957: Unable to initialize the key store. (CONNECTION_ID=G84MqcYmRTKVyVYhQmbzyw==)
SSO not found
SSO not found
SSO KeyStore not available
SSO KeyStore not available

Could you please guide me in resolving this issue?

Your prompt assistance in this matter is highly appreciated.

Best regards,
Desy

### DBeaver Version

Community Edition 23.3.1

### Operating System

Windows 11 Home Single Language

### Database and driver

RDS Oracle Standard Edition Two 19.0.0.0.ru-2023-01.rur-2023-01.r1

### Steps to reproduce

1. Create an Oracle connection with TNS connection type.
2. In the Driver Properties tab, fill in the oracle.net.wallet_location parameter with the file located on your laptop: (SOURCE=(METHOD=FILE)(METHOD_DATA=(DIRECTORY=C:\Program Files\Oracle\instantclient_21_9\network\admin\cwallet.sso))).

**Comment by LonwoLonwo** [comment]
Hello @dsyayu 

I'm not sure that your path to your wallet is correct.
You have the `DIRECTORY=` property and then the specific file path.

`-Doracle.net.wallet_location= “(SOURCE=(METHOD=FILE)(METHOD_DATA=(DIRECTORY=/users/test/wallet_dbname)))” `

Here is an example from the [documentation](https://docs.oracle.com/en/cloud/paas/autonomous-database/serverless/adbsb/connect-jdbc-thin-wallet.html#GUID-32A48CAA-89AC-40A4-AFD1-BB962C562805).

Also, you can use the wallet path in [the connection URL string](https://docs.oracle.com/en/database/oracle/oracle-database/21/jjdbc/data-sources-and-URLs.html#GUID-843F6EC2-EDFD-4A88-98DE-B45E1CEADC0E).

**Comment by LonwoLonwo** [comment]
Additionally some tips you can find here: https://github.com/dbeaver/dbeaver/issues/6912

**Comment by E1izabeth** [comment]
It's been a while since no update here. If the issue is still actual please let me know and provide additional information.

**Comment by asgh** [comment]
@dsyayu  You need to manually add the `com.oracle.database.security:oraclepki` Library.

@E1izabeth maybe that library should be part of the default set of Oracle libraries.

**Comment by E1izabeth** [comment]
@dsyayu does it help?

**Comment by toni3520** [comment]
I agreed with @asgh `com.oracle.database.security:oraclepki` should be default library otherwise user needs to find this GitHub issue. Maybe can detect if SSO not found then dbeaver can show dialog to add the library

**Comment by E1izabeth** [comment]
@asgh @toni3520 thanks for the tip

**Comment by ShadelessFox** [comment]
Hello @dsyayu @asgh @toni3520,

We have Oracle Wallet support in PRO products. I would like to kindly ask you to get a trial version and see if it works there:
https://dbeaver.com/trial/

I'm asking because we haven't received any reports of Wallet authentication not working by our commercial users. I don't quite understand why this is the case. I have a suspicion that it might be due to the presence of the BouncyCastle library there.

If it works for you, we will add the mentioned dependency just to the community edition of DBeaver.

**Comment by E1izabeth** [comment]
It's been a while since no update here. If the issue is still actual please let me know and provide additional information.

**Comment by ultradumb** [comment]
The issue re-appeared in dBeaver Version 25.2.4.202511021717. I do not remember what version was there before I updated few days ago.
It seems that certain versions of dBeaver included the file oraclepki.jar in the driver, while some others did not, as this issue re-appeared few times after updates, and it re-appeared now again. I had to manually include the oraclepki.jar in the driver 'Libraries' tab, as seen on the attached pic.
Note: I did not change anything in my Oracle Instant Client (version 21) installation for last 2 years. The only place I can see oraclepki.jar on my PC is in c:\oracle\sqlcl\lib directory, so it was never present in Oracle Instant Client. It had to be in DBeaver Driver set then.

<img width="438" height="206" alt="Image" src="https://github.com/user-attachments/assets/48d83ed3-944f-4ccf-a7e5-92d714710d6e" />
